packages/vitest/src/utils/coverage.ts

```diff
@@ -272,6 +273,37 @@ export class BaseCoverageProvider {
       return chunks
     }, [])
   }
+
+  createUncoveredFileTransformer(ctx: Vitest) {
+    const servers = [
+      ...ctx.projects.map(project => ({
+        root: project.config.root,
+        vitenode: project.vitenode,
+      })),
+      // Check core last as it will match all files anyway
+      { root: ctx.config.root, vitenode: ctx.vitenode },
+    ]
+
+    return async function transformFile(filename: string) {
+      let lastError
+
+      for (const { root, vitenode } of servers) {
+        if (!filename.startsWith(root)) {
+          continue
+        }
+
+        try {
+          return await vitenode.transformRequest(filename)
+        }
+        catch (error) {
+          lastError = error
+        }
+      }
+
+      // All vite-node servers failed to transform the file
+      throw lastError
```

Comment:
Hm, what if the file never started with root? Is it possible? Then this will be undefined

Reply:
Oh good point! This might fail with allowExternal files. Ill need to add test case for those.

Reply:
Opened https://github.com/vitest-dev/vitest/pull/6372 as follow-up. I didn't yet look into this a lot, created PR mostly so that I won't forget this issue.